nvme.c

```diff
@@ -5317,6 +5321,330 @@ static int show_registers(int argc, char **argv, struct command *cmd, struct plu
 	return err;
 }
 
+bool nvme_is_fabrics_reg(int offset)
+{
+	switch (offset) {
+	case NVME_REG_CAP:
+		fallthrough;
+	case NVME_REG_VS:
+		fallthrough;
+	case NVME_REG_CC:
+		fallthrough;
+	case NVME_REG_CSTS:
+		fallthrough;
+	case NVME_REG_NSSR:
+		fallthrough;
+	case NVME_REG_CRTO:
+		return true;
+	default:
+		break;
+	}
+
+	return false;
+}
+
+static int get_register_property(int fd, void **pbar, int offset)
+{
+	int err, size = getpagesize();
+	__u64 value;
+	void *bar = malloc(size);
```

Comment:
Is it necessary to allocate a full page of memory just to store a single property?

Reply:
Since the caller function uses the output paramter pbar with the offset to shift the allocated memory as same with the bar memory mapped by the function mmap_registers() to use so necessary to allocate a full page memory as mentioned.